### PR TITLE
Make exclude_border independent and anisotropic

### PR DESCRIPTION
**Pull request by ttung**
## Description

Currently, `exclude_border` is only tested for truthiness when a footprint is defined.  In reality, it should be independent and anisotropic.  This PR makes it both.

This adds tests to verify that `exclude_border` behaves as expected.

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [x] Unit tests
- [x] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by ttung** [comment]
The blobs test takes an image, makes many versions of it with varying degrees of gaussian blur.  It calls `peak_local_max` finder with exclude_border=5, expecting to remove 5 px from each dimension (which would be wrong even if `peak_local_max` interpreted exclude_border=5 correctly), but in fact was just removing pixels based on the footprint.

Prior to any change, it was removing 1 pixel across each of the dimensions closest to the origin, and 2 pixels across each of the dimensions farthest from the origin.  This happened to clip out the peak.

After #4324, it was removing 1 pixel across each of the dimensions closest to the origin, and 1 pixel across each of the dimensions farthest from the origin.  This happened to not clip out the peak, and reported a spot.

With this PR, it removes 5 pixels from each of the dimensions closest to the origin, and 5 pixels across each of the dimensions farthest from the origin.  This happened to clip out the peak.

The actual correct behavior requires modifying the blobs code, which I will do.

**Comment by ttung** [comment]
See https://github.com/scikit-image/scikit-image/pull/4327 :)

File: skimage/feature/peak.py [hunk 1]
```diff
@@ -85,9 +84,13 @@ def peak_local_max(image, min_distance=1, threshold_abs=None,
...
         Minimum intensity of peaks, calculated as `max(image) * threshold_rel`.
-    exclude_border : int or bool, optional
-        If nonzero int, `exclude_border` excludes peaks from
-        within `exclude_border`-pixels of the border of the image.
+    exclude_border : tuple of ints or bool, optional
+        If tuple of ints, the length of the tuple must match the input array's
+        dimensionality.  Each element of the tuple will exclude peaks from
+        within `exclude_border`-pixels of the border of the image along that
+        dimension.
+        If nonzero int, `exclude_border` excludes peaks from within
```
Comment (emmanuelle) [inline skimage/feature/peak.py]: Could you please put the "nonzero int" case first (before "tuple of ints") since I would expect it to be more common?

Comment (ttung) [reply skimage/feature/peak.py]: Updated.

**Comment by ttung** [comment]
I added some test cases for illegal values (negative integers, and tuples containing negative integers).

**Comment by emmanuelle** [comment]
Thanks a lot @ttung, merging!

**Comment by ttung** [comment]
Any chance https://github.com/scikit-image/scikit-image/pull/4327 can also get reviewed? :)

**Pull request merged**